### ISSUE TITLE
Make celery workers to read DAG from the main host (scheduler)

### ISSUE THREAD

**Issue by AlekseiSaff** [apache/airflow#28371]
### Description

I'd like not to sync/NFS mount DAG folder to workers. Instead workers should have an ability to read DAGs from scheduler or from DB

### Use case/motivation

less maintenance and avoid mistakes

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
This is not how Airflow works in a distributed way. The architecture of Airflow is very  explicit and specific about that. Airlow does not take care about DAG distribution on its own - it is delegated to the appropriate deployment mechanisms. We simply do not want to implement somethign that other, existing mechanisms are dedicated to (and they are much better than Airflow in doing so). 
 
This is a fundamental change in the architecture and if you would like to pursue that idea (though I can tell you, it's very unlikely to happen), Feature request is not nearly enough. You would have to start a devlist discussion, prepare a complete Airflow Improvoment Proposal and convince others that it makes sense (that should include a detailed proposal on how you want to solve all the cases of distributed environment challenges). See https://airflow.apache.org/community/ tor the relevant links for the devlist and https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals for the proposals. 

Closing this as "no implement" - but if you are not discouraged - follow the path I explained.

**Comment by potiuk** [comment]
This is not how Airflow works in a distributed way. The architecture of Airflow is very  explicit and specific about that. Airlow does not take care about DAG distribution on its own - it is delegated to the appropriate deployment mechanisms. We simply do not want to implement somethign that other, existing mechanisms are dedicated to (and they are much better than Airflow in doing so). 
 
This is a fundamental change in the architecture and if you would like to pursue that idea (though I can tell you, it's very unlikely to happen), Feature request is not nearly enough. You would have to start a devlist discussion, prepare a complete Airflow Improvoment Proposal and convince others that it makes sense (that should include a detailed proposal on how you want to solve all the cases of distributed environment challenges). See https://airflow.apache.org/community/ tor the relevant links for the devlist and https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals for the proposals. 

Closing this as "no implement" - but if you are not discouraged - follow the path I explained.
